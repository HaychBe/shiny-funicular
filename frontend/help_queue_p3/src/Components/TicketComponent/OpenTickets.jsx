import Card from 'react-bootstrap/Card';
import DeleteTicket from './DeleteTicket';
import UpdateTicket from './UpdateTicket';
import ViewTickets from './ViewTickets';

const OpenTickets = ({openTicket}) => {
    // console.log(openTicket);

    return(
        
        <Card style={{ width: '18rem' }}>
    <Card.Body>
        <Card.Title>{openTicket.title}</Card.Title>
        <Card.Subtitle className="mb-2 text-muted">{openTicket.trainee}</Card.Subtitle>
        <Card.Text>
            <small className="text-muted">Logged: {openTicket.dateTime}</small>
        </Card.Text>
        <ViewTickets
            id={openTicket.id}
            title={openTicket.title}
            traineeName={openTicket.trainee}
            trainer={openTicket.trainer}
            cohort={openTicket.cohort}
            description={openTicket._description}
            topic={openTicket.topic}
            completed={openTicket.completed}
            />
        <UpdateTicket
            id = {openTicket.id}
            oldTitle = {openTicket.title}
            oldTraineeName = {openTicket.title}
            oldTrainer = {openTicket.trainer}
            oldCohort = {openTicket.cohort}
            oldDescription = {openTicket._description}
            oldTopic = {openTicket.topic}
            />
        <DeleteTicket
            id = {openTicket.id}
            />
    </Card.Body>
</Card>

)
}

export default OpenTickets;