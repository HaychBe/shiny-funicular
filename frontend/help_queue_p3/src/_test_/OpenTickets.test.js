import React from 'react';
import renderer from 'react-test-renderer';
import OpenTickets from  '../Components/TicketComponent/OpenTickets';





it('renders correctly', () => {
    const title = "Jest ";
    const topic = "Jest Test ";
    const traineeName = "Shamsi";
    const cohort = "cne";
    const trainer = "Reece";
    const description = "Jest Test description";
    const time = "0505";
    const tree = renderer.create(<OpenTickets title={title} topic={topic} traineeName={traineeName} cohort={cohort} trainer={trainer} description={description} time={time}/>).toJSON();
    expect(tree).toMatchSnapshot();
});


// import React from 'react';
// import renderer from 'react-test-renderer';
// import CreateTicket from  '../Components/TicketComponent/CreateTicket';
// import ClosedTickets from  '../Components/TicketComponent/ClosedTickets';





// it('renders correctly', () => {
//     const id = 2;
//     const title = "Jest ";
//     const topic = "Jest Test ";
//     const traineeName = "Shamsi";
//     const cohort = "cne";
//     const trainer = "Reece";
//     const description = "Jest Test description";
//     const time = "0505";
//     // const completed = true;
//     const tree = renderer.create(<ClosedTickets id={id} Title={title} topic={topic} traineeName={traineeName} cohort={cohort} trainer={trainer} description={description} time={time} />).toJSON();
//     expect(tree).toMatchSnapshot();
// });

