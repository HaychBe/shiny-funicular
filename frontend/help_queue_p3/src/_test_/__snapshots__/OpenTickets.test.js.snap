// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpenTicket should match the snapshot every time it is rendered 1`] = `
<button
  className="btn btn-success"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Create Ticket
</button>
`;

exports[`renders correctly 1`] = `
<button
  className="btn btn-success"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Create Ticket
</button>
`;
